woocommerce/woocommerce-ios
# [Woo POS] Update card reader connection to integrate with the payment service in the pre-existing design

Part of #12916 


## Why

Now that we have a payment service with a working card reader connection https://github.com/woocommerce/woocommerce-ios/pull/12909, this PR integrates the existing entry in the navigation bar with the payment service. The entry point might move to a menu p1717083791879079-slack-C070SJRA8DP, and we'll update the UI & onboarding/connection alerts UI separately.

## How

Now that we're reusing the card connection flow in the payment flow, `CardReaderConnectionView` was replaced with `CardReaderConnectionStatusView` which is a simple status view that indicates whether a card reader is connected as in the latest design (battery & name of the reader isn't shown). `CardReaderConnectionViewModel` was updated to take in a `CardPresentPaymentFacade`, and provide the connection status based on the payment service. Then, the card reader connection status view is shown in the navigation bar in `PointOfSaleDashboardView`.

## Testing information


Prerequisite: having a store eligible for POS

- Switch to a store in the prerequisite if needed
- Turn on the feature switch in Menu > Settings > Experimental Features > POS, if needed
- Go to the Menu tab --> the POS row should appear shortly
- Tap `Reader disconnected` in the navigation bar (the connection will be checked beforehand in a future task after we decide on the menu design) --> a modal should be shown about the reader connection
- Since there's no CTA in the alert UI, dismiss the alert --> the navigation bar should show `Reader connected` as the placeholder implementation

## Screenshots


In the screencast below, stimulated card reader configuration was enabled for testing purposes.

https://github.com/woocommerce/woocommerce-ios/assets/1945542/7a16488a-4f8a-4d66-b741-8b7c1f3ebb83



---
